Firestore/Source/Model/FSTMutationBatch.mm

```diff
@@ -116,6 +133,22 @@ - (FSTMaybeDocument *_Nullable)applyToLocalDocument:(FSTMaybeDocument *_Nullable
   return maybeDoc;
 }
 
+- (FSTMaybeDocumentDictionary *)applyToLocalDocumentSet:(FSTMaybeDocumentDictionary *)documentSet {
+  // TODO(mrschmidt): This implementation is O(n^2). If we iterate through the mutations first (as
+  // done in `applyToLocalDocument:documentKey:`), we can reduce the complexity to O(n).
+
+  FSTMaybeDocumentDictionary *mutatedDocuments = documentSet;
+  for (FSTMutation *mutation in self.mutations) {
+    const DocumentKey &key = mutation.key;
+    FSTMaybeDocument *mutatedDocument =
+        [self applyToLocalDocument:[mutatedDocuments objectForKey:key] documentKey:key];
+    if (mutatedDocument) {
+      mutatedDocuments = [mutatedDocuments dictionaryBySettingObject:mutatedDocument forKey:key];
```

Comment:
Is there a mutable version of `FSTMaybeDocumentDictionary`? Would it be more efficient?